### PR TITLE
Task 2 (Connecting app to "Foreca" weather api)

### PR DESCRIPTION
**Pull request by StefanAtanasov98**
Connecting the application to "Foreca" weather api includes:

- Making a simple server;
- Using AJAX to make a request for non-expiring token;
- Using AJAX to make a request for filtered list of cities;

image

### CODE REVIEW

File: src/components/Homepage/Nav/DropdownMenu.js [hunk 1]
```diff
@@ -27,11 +30,13 @@ function DropdownMenu(props) {
   const filteredCities = searchedCity.name
     ? cities.filter(city => city.name.includes(searchedCity.name))
     : cities;
+
```
Comment (virzhinia-petkova-redshelf) [inline src/components/Homepage/Nav/DropdownMenu.js]: do you still need to filter cities?

File: src/customHooks/useAuthorize.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react';
+
+const URL = 'http://localhost:5020/auth';
+
+const useAuthorize = () => {
+  useEffect(() => {
+    (async () => {
+      if (!localStorage.getItem('token')) {
```
Comment (virzhinia-petkova-redshelf) [inline src/customHooks/useAuthorize.js]: you have a string that you're using in different places in the app. It's very easy to do a typo and spend too much time wondering what's happening and what broke. Please move it to a constant you can use throughout the app.

File: src/styles/DropdownMenu.module.css [hunk 3]
```diff
@@ -8,9 +8,10 @@
 .dropdownContent {
   display: inline-block;
   background-color: var(--white);
-  position: absolute;
   top: 50px;
   box-shadow: 0 8px 16px 0 var(--black);
+  max-height: 250px;
+  overflow: auto;
 }
```
Comment (virzhinia-petkova-redshelf) [inline src/styles/DropdownMenu.module.css:18]: where did you fix your footer to stick to the bottom of the page?

Comment (StefanAtanasov98) [reply src/styles/DropdownMenu.module.css:18]: I've set the Footer's properties "position: absolute" with "bottom: 0" and I added "position: relative" to the dropdown div. Before I made the dropdown content to display in a shorter list I tested the footer and it stuck to the bottom all the way down the cities list.

Comment (StefanAtanasov98) [reply src/styles/DropdownMenu.module.css:18]: I also added "position: relative", "min-height: 100vh" and "margin: 0" to the html body tag

File: src/components/Homepage/Nav/DropdownMenu.js [hunk 4]
```diff
@@ -23,15 +27,13 @@ function DropdownMenu(props) {
   const onSearch = ev => {
     setSearchedCity({ name: ev.target.value });
   };
-
-  const filteredCities = searchedCity.name
-    ? cities.filter(city => city.name.includes(searchedCity.name))
-    : cities;
-  const citiesList = filteredCities.map(city => (
+  
+  const citiesList = cities.sort((a, b) => a.country.localeCompare(b.country)).map(city => (
```
Comment (virzhinia-petkova-redshelf) [inline src/components/Homepage/Nav/DropdownMenu.js]: aren't the cities coming already sorted?

File: src/customHooks/useAuthorize.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react';
+
+const URL = 'http://localhost:5020/auth';
+const token = 'token';
```
Comment (virzhinia-petkova-redshelf) [inline src/customHooks/useAuthorize.js]: 1. the convention for constants is UPPERCASE, so TOKEN
2. is this the best place to store a global variable for your app?
3. you can simply to `export const TOKEN `here

File: src/styles/DropdownMenu.module.css (hunk 3 above)
Comment (virzhinia-petkova-redshelf) [reply src/styles/DropdownMenu.module.css:18]: what do you need the position relative of the dropdown div for?

Comment (StefanAtanasov98) [reply src/styles/DropdownMenu.module.css:18]: Well, I had problems positioning the "Cities" button itself and the dropdown content, for example when my content appears the button shrunk to half its height, so I tried a lot of things and this is what I managed. Now, when I remove "position: relative" from dropdown div, the dropdown content doesn't appear on the screen when I click the button.

File: src/components/Homepage/Nav/DropdownMenu.js (hunk 4 above)
Comment (StefanAtanasov98) [reply src/components/Homepage/Nav/DropdownMenu.js]: They aren't.. Should they be?

File: src/customHooks/useAuthorize.js (hunk 5 above)
Comment (StefanAtanasov98) [reply src/customHooks/useAuthorize.js]: 2. Maybe in a separate file just for global variables..?

File: src/styles/DropdownMenu.module.css (hunk 3 above)
Comment (virzhinia-petkova-redshelf) [reply src/styles/DropdownMenu.module.css:18]: okay

File: src/components/Homepage/Nav/DropdownMenu.js (hunk 4 above)
Comment (virzhinia-petkova-redshelf) [reply src/components/Homepage/Nav/DropdownMenu.js]: I guess I thought they were. No changes needed.

File: src/customHooks/useAuthorize.js (hunk 5 above)
Comment (virzhinia-petkova-redshelf) [reply src/customHooks/useAuthorize.js]: sounds like a plan

**Review by virzhinia-petkova-redshelf — APPROVED** [review]

**Pull request merged**